src/credentials/FabricTable.cpp

```diff
@@ -1788,15 +1788,18 @@ CHIP_ERROR FabricTable::CommitPendingFabricData()
         CHIP_ERROR err = CHIP_NO_ERROR;
         if (hasInvalidInternalState)
         {
-            ChipLogError(FabricProvisioning, "Failed to commit: internally inconsistent state!");
-            err = CHIP_ERROR_INTERNAL;
-        }
-        else if (haveNewTrustedRoot)
-        {
-            ChipLogError(FabricProvisioning,
-                         "Failed to commit: tried to commit with only a new trusted root cert. No data committed.");
-            hasInvalidInternalState = true;
-            err                     = CHIP_ERROR_INCORRECT_STATE;
+            if (haveNewTrustedRoot)
+            {
+                ChipLogError(FabricProvisioning,
+                             "Failed to commit: tried to commit with only a new trusted root cert. No data committed.");
```

Comment:
Hmm. Previously this message was printed if `!hasInvalidInternalState` which implied `!hasPending`. Is it still correct to say "with only a new trusted root cert" after you changed this code to be called if `hasInvalidInternalState`?

Reply:
Yes. The `hasInvalidInternalState` accounts for this.

Reply:
My point is that `hasInvalidInternalState` can be set for multiple reasons and the error message suggests that the problem was that someone tried to commit a new trusted root certificate alone (without any other changes). Did I get it right? If I did, I think that the existing conditions are not sufficient.

Reply:
Addressed with clearer logic